### PR TITLE
fix(scrollviewer): fix scrollability and offset calculations

### LINKED ISSUE #8947: [Gallery][Skia.GTK] `Chip` - Inside the `XamlDisplay` control area, when scrolling up, all the chip moves to the left side and the chip gets invisible
### Current behavior

Image

https://user-images.githubusercontent.com/74563602/171275745-34beed72-edda-4981-b386-945512711c0d.mp4

### Expected behavior

The Chip inside the layout should not move when scrolling up and down.
https://user-images.githubusercontent.com/74563602/171276331-36c138eb-bd31-4f75-8855-f52d47803b54.mp4

### How to reproduce it (as minimally and precisely as possible)

1. Open the link https://unogallerycanary.z27.web.core.windows.net/
2. Change theme from Day mode to Night mode.
3. Click/tap Burger Menu on the left top right.
4. click/tap "Chip" from the "Toolkit" list.
5. Click inside any chip XamlDisplay control area ( for example Filter)
6. Scroll up using a mouse, observe that all the chip moves to the left side and the chip gets invisible.

### Workaround

_No response_

### Works on UWP/WinUI

_No response_

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

Skia (GTK on Linux/macOS/Windows)

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #8947

## PR Type

What kind of change does this PR introduce?
- Bugfix

### <samp>🤖 Generated by Copilot at c72cb35</samp>

This pull request fixes two bugs and adds one test related to the `ScrollContentPresenter` and the `ScrollViewer` controls. It makes the `ScrollContentPresenter` respect the scroll mode settings and the mouse wheel input, and it verifies that the `ScrollViewer` does not scroll when the content fits the viewport. The changes affect the files `ScrollContentPresenter.Managed.cs`, `ScrollContentPresenter.cs`, and `Given_ScrollViewer.cs`.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.cs [hunk 1]
```diff
@@ -200,22 +200,19 @@ private void PointerWheelScroll(object sender, Input.PointerRoutedEventArgs e)
 				{
 					// TODO: Handle zoom https://github.com/unoplatform/uno/issues/4309
 				}
-				else if (!canScrollVertically || properties.IsHorizontalMouseWheel || e.KeyModifiers == VirtualKeyModifiers.Shift)
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.cs:203]: Is the Shift modifier not needed anymore?

Comment (ramezgerges) [reply src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.cs:203]: I'm not sure. I tested on WinUI and shift doesn't seem to do anything either horizontally or vertically, so I'm not sure why that was in there in the first place. A quick search also didn't yield anything.

Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.cs:203]: If you enable vertical **and** horizontal scrolling, holding the shift key will allow you to scroll horizontally with the vertical wheel of you mouse. This is a common usage, not only with WinUI but also windows, web browsers etc.

File: src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.cs [hunk 2]
```diff
@@ -200,22 +200,19 @@ private void PointerWheelScroll(object sender, Input.PointerRoutedEventArgs e)
...
+				else if (canScrollHorizontally && properties.IsHorizontalMouseWheel)
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.cs:203]:
```suggestion
				else if (canScrollHorizontally && (!canScrollVertically || properties.IsHorizontalMouseWheel || e.KeyModifiers == VirtualKeyModifiers.Shift))
```

**Review by dr1rrb — APPROVED** [review]

**Pull request merged**